File: apps/user-web/src/types/notification.ts
// 通知类型定义

export interface Notification {
  id: string
  type: 'milestone' | 'task_review' | 'income' | 'system' | 'group' | 'course'
  title: string
  content: string
  read: boolean
  link?: string
  createdAt: string
}

export interface NotificationSettings {
  taskNotification: boolean
  bbsNotification: boolean
  incomeNotification: boolean
  systemNotification: boolean
}

File: server/blueblood-api/src/main/java/com/blueblood/api/modules/notification/entity/Notification.java
package com.blueblood.api.modules.notification.entity;

import com.baomidou.mybatisplus.annotation.TableName;
import com.blueblood.api.common.entity.BaseEntity;
import lombok.Data;
import lombok.EqualsAndHashCode;

/**
 * 通知主表（按用户维度，user_id 为接收用户）。
 */
@Data
@EqualsAndHashCode(callSuper = true)
@TableName("notification")
public class Notification extends BaseEntity {

    /** 接收用户ID */
    private Long userId;

    /** 类型：milestone/task_review/income/system/group/course */
    private String type;

    private String title;

    private String content;

    /** 跳转链接 */
    private String link;
}
